src/components/patch-assignment-form/components/confirm-reassignment-dialog.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { Dispatch, SetStateAction } from "react";
+
+import { switchPatchAssignments } from "../utils/switch-patch-assignments";
+
+import { Patch } from "@mtfh/common/lib/api/patch/v1";
+import { Button, Dialog, DialogActions, Link } from "@mtfh/common/lib/components";
+
+interface ConfirmReassignmentDialogProps {
+  onSuccess: Function;
+  onDialogCancel: Function;
+  setRequestError: Dispatch<SetStateAction<string | null>>;
+  isOpen: boolean;
+  reassigningPatch: Patch | null;
+  switchingWithPatch: Patch | null;
+}
+
+export const ConfirmReassignmentDialog = ({
+  onSuccess,
+  onDialogCancel,
+  setRequestError,
+  isOpen,
+  reassigningPatch,
+  switchingWithPatch,
+}: ConfirmReassignmentDialogProps): JSX.Element => {
+  if (!reassigningPatch || !switchingWithPatch) return <></>;
+  const onSwitchAssignmentSuccess = () => {
+    onSuccess();
+  };
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onDismiss={() => {
+        onDialogCancel();
+      }}
+      title="Switch assignment"
+    >
+      <p data-testid={`reassign-message-${switchingWithPatch.name}`}>
+        Reassigning <strong>{reassigningPatch?.responsibleEntities[0].name}</strong> to{" "}
+        <strong>{switchingWithPatch?.name}</strong>
+      </p>
+      <p data-testid={`reassign-message-${reassigningPatch.name}`}>
+        Reassigning <strong>{switchingWithPatch?.responsibleEntities[0].name}</strong> to{" "}
+        <strong>{reassigningPatch?.name}</strong>
+      </p>
```

Comment:
Just a suggestion (of which im not sure is preferable here). You could pass this logic as children, and avoid passing additonal props ([prop drilling](https://kentcdodds.com/blog/prop-drilling)).  

It would look something like

```
<ConfirmReassignmentDialog {...otherProps}>
<p data-testid={`reassign-message-${switchingWithPatch.name}`}>
        Reassigning <strong>{reassigningPatch?.responsibleEntities[0].name}</strong> to{" "}
        <strong>{switchingWithPatch?.name}</strong>
      </p>
      <p data-testid={`reassign-message-${reassigningPatch.name}`}>
        Reassigning <strong>{switchingWithPatch?.responsibleEntities[0].name}</strong> to{" "}
        <strong>{reassigningPatch?.name}</strong>
      </p>
</ConfirmReassignmentDialog>
```